src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,32 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() != toX) {
+            setHorizontalDirection(robot, toX);
+            while (robot.getX() != toX) {
+                robot.stepForward();
+            }
+        }
+        if (robot.getY() != toY) {
+            setVerticalDirection(robot, toY);
+            while (robot.getY() != toY) {
+                robot.stepForward();
+            }
+        }
+    }
+
+    private void setHorizontalDirection(Robot robot, int toX) {
+        Direction neededDirection = robot.getX() < toX ? Direction.RIGHT : Direction.LEFT;
+        while (!robot.getDirection().equals(neededDirection)) {
+            robot.turnLeft();
+        }
+    }
+
+    private void setVerticalDirection(Robot robot, int toY) {
+        Direction neededDirection = robot.getX() < toY ? Direction.UP : Direction.DOWN;
+        while (!robot.getDirection().equals(neededDirection)) {
+            robot.turnLeft();
+        }
     }
```

Comment:
can we combine these methods into one?

Reply:
I combined these two methods into one with switch -case